File: app/src/main/java/id/ac/polinema/seameo/ecanteen/model/TransactionModel.java
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 6/21/19 12:14 PM
 */

package id.ac.polinema.seameo.ecanteen.model;

import java.util.ArrayList;

public class TransactionModel {
    public static String NAME = "name";
    public static String DATE_TIME = "date_time";
    public static String ITEMS = "items";
    public static String MONEY = "money";
    public static String CASHBACK = "cashback";
    public static String PAYMENT = "payment";

    private String name;
    private String dateTime;
    private ArrayList<String> items;
    private int money;
    private int cashback;
    private int payment;

    public ArrayList<String> getItems() {
        return items;
    }

    public void setItems(ArrayList<String> items) {
        this.items = items;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDateTime() {
        return dateTime;
    }

    public void setDateTime(String dateTime) {
        this.dateTime = dateTime;
    }

    public int getMoney() {
        return money;
    }

    public void setMoney(int money) {
        this.money = money;
    }

    public int getCashback() {
        return cashback;
    }

    public void setCashback(int cashback) {
        this.cashback = cashback;
    }

    public int getPayment() {
        return payment;
    }

    public void setPayment(int payment) {
        this.payment = payment;
    }
}

File: app/src/main/java/id/ac/polinema/seameo/ecanteen/model/TransactionModel.kt
/*
 * Copyright (c) 2019.
 *
 * Author: Mahatta Maulana
 * Github: https://github.com/hattamaulana
 *
 * Last Modified at 9/27/19 10:08 PM
 */

package id.ac.polinema.seameo.ecanteen.model

data class TransactionModel (
        var uid: String = "",
        var name: String = "",
        var dateTime: String = "",
        var orders: List<OrderModel>? = null,
        var noTable: Int = -1,
        var money: Int = 0,
        var cashback: Int = 0,
        var nominal: Int = 0
) {
    fun toMap() = mapOf<String, Any?>(
            "uid" to uid,
            "name" to name,
            "dateTime" to dateTime,
            "orders" to orders,
            "noTable" to noTable,
            "money" to money,
            "cassback" to cashback,
            "money" to nominal
    )
}
